docs/source/kinds_of_types.rst

```diff
@@ -188,6 +188,130 @@ using bidirectional type inference:
 If you want to give the argument or return value types explicitly, use
 an ordinary, perhaps nested function definition.
 
+Extended Callable types
+***********************
+
+As an experimental mypy extension, you can specify ``Callable`` types
+that support keyword arguments, optional arguments, and more.  The
+"argument" part of the ``Callable`` specification (the part specifying
```

Comment:
Where you specify the arguments of a ``Callable``, you can choose to supply just the type of a nameless positional argument, or an "argument specifier" representing a more complicated form of argument.